core/modules/rproc/InfileMerger.h

```diff
@@ -156,10 +157,18 @@ class InfileMerger {
     std::mutex _createTableMutex; ///< protection from creating tables
     std::mutex _sqlMutex; ///< Protection for SQL connection
     bool _needCreateTable{true}; ///< Does the target table need creating?
+    size_t _getResultTableSizeMB(); ///< Return the size of the result table in MB.
 
     mysql::MySqlConnection _mysqlConn;
+
     std::mutex _mysqlMutex;
     lsst::qserv::mysql::LocalInfile::Mgr _infileMgr;
+
+    std::string _queryIdStr{"QI=?"}; ///< Unknown until results start coming back from workers.
+
+    int _sizeCheckRowCount{0}; ///< Number of rows read since last size check.
+    int _checkSizeEveryXRows{1000}; ///< Check the size of the result table after every x number of rows.
```

Comment:
I'd probably use much higher number for that. I don't think we need very high precision for cutoff, checking every 100MB is probably OK. With record sizes at roughly 1KB it should be OK to check every 100,000 rows. Would be interesting to see if we can guess result row size, and increase check interval it further for short rows to reduce overhead.